app/services/fixity_checker.rb

```diff
@@ -0,0 +1,117 @@
+# FixityChecker.new(asset).check
+# FixityChecker.new(asset).prune_checks
+# This first version of the fixity checker only checks
+# for actual discrepancies between already-recorded
+# checksums and calculated ones.
+#
+# This means:
+# Assets with nil files are ignored.
+# Files with nil checksums are ignored.
+
+class FixityChecker
+  attr_reader :asset
+
+  NUMBER_OF_RECENT_PASSED_CHECKS_TO_KEEP = 5
+  CHECK_CYCLE_LENGTH = 7
+
+
+  # @param work [Work] Work object, it's members will be put into a zip
+  # @param callback [proc], proc taking keyword arguments progress_i: and progress_total:, can
+  #   be used to update a progress UI.
+  def initialize(asset)
+    raise ArgumentError.new("Please pass in an Asset.") unless asset.is_a? Asset
+    @asset = asset
+  end
+
+  #FixityChecker.new(asset).check
+  def check(sieve_integer = 0)
+    return unless sift(sieve_integer)
+    return nil if @asset.nil?
+    return nil if @asset.file.nil?
+    the_expected_checksum = expected_checksum
+    # Note: @asset.file.sha512 can be nil on e.g.
+    # items imported with DISABLE_BYTESTREAM_IMPORT=true
+    # For now, we are ignoring these.
+    return nil if the_expected_checksum.nil?
```

Comment:
I think it should raise an exception if it can't fixity check, rather than silently ignore/fail. Fixity checking too important to make it that easy to not know you didn't do one. The caller would be responsible for not calling `check` unless it can be checked. 

I think that applies to the @asset.nil? and @asset.file.nil? checks above too -- they should just raise. Maybe actually check the @asset argument in the initializer, and raise argument error there if you got nil or an asset that can't be fixity checked?

Ah, I see the problem though, then the rake task has to know if an asset can be fixity checked or not before trying... But in fact, any asset that is `stored?` (has had ingest finalized) ought to be able to be checked -- if something that is `stored?` can't be checked, we want to find out about it one way or another, it's a software problem, we don't want it to be silently skipped and not checked. So rake task could just check `stored?` maybe.

Reply:
Great advice.

Reply:
Modified rake task to check whether the asset is `stored?` before checking it, and added corresponding exceptions.